onnx/__init__.py

```diff
@@ -153,6 +174,14 @@ def save_model(proto, f, format=None):  # type: (Union[ModelProto, bytes], Union
     f can be a file-like object (has "write" function) or a string containing a file name
     format is for future use
     '''
+    if isinstance(proto, bytes):
+        proto = _deserialize(proto, ModelProto())
+
+    model_filepath = _get_file_path(f)
+    if model_filepath:
+        basepath = os.path.dirname(model_filepath)
+        proto = write_external_data_tensors(proto, basepath)
+
```

Comment:
We can add a parameter in save_model, all_data_as_external=False, so all tensors' data will be save to external_data. We probably need to modify the proto, and revert it back after saving.

Reply:
how about having three options for users: 
1) embedded weights (default)
2) one file for all external data
3) separate files for different external data

Reply:
Added a helper convert_model_to_external_data. Users can call this function first and then save tensor data externally.